app/views/accounts/show.html.slim

```diff
@@ -57,4 +57,4 @@ h1.hide = t('titles.account')
     = t('headings.account.account_history')
     = image_tag asset_url('history.svg'), width: 12, class: 'ml1'
   - @view_model.recent_events.each do |event|
-    = render @view_model.recent_event_partial, event: event
+    = render event.event_partial, event: event
```

Comment:
What's everyone else's thought on doing something like this instead:

```slim
- if event.is_a? EventDecorator
  render 'event_partial', event: event
- elsif event.is_a? IdentityDecorator
  render 'identity_partial', identity: identity
```

Reply:
Why not use the existing `decorated_session` helper? `decorated_session.sp_return_url` https://github.com/18F/identity-idp/blob/master/app/decorators/service_provider_session_decorator.rb#L62-L68

Reply:
We could, that involves a little bit more refactoring here since `identity.service_prover` returns a string (the sp's issue) and not a ServiceProvider object. That was my reason for reading it from the metadata.

Reply:
Although we might need to add a new method that only calls `sp.return_to_sp_url` because the current `sp_return_url` modifies the redirect URL to an access denied URL for OIDC. I'm not sure why we do that. Do you remember why @zachmargolis ?

Reply:
Also, we should enforce the presence of `return_to_sp_url` on the Dashboard. Many SPs don't have that filled in according to `service_providers.yml`

Reply:
@monfresh hmm, it seems kind of weird to use the session decorator since it looks like it was meant to decorate the session, and not a ServiceProvider in isolation.

Reply:
Also, I think we modify it to be an access denied URL since it is used to redirect back to the SP after we fail to verify someone?

Reply:
No, the way we're currently using it is for all links that go back to the SP, such as on the branded page and on the sign in page and some IdV pages, which sounds wrong to me if they point is to simply go back to the SP.

Reply:
I just had a conversation with Jonathan and I did not understand this was meant as a permanent link to any SP you might have visited in the past, and not tied to the one in the current session. So, never mind what I said about session_decorator.

Reply:
> Although we might need to add a new method that only calls sp.return_to_sp_url because the current sp_return_url modifies the redirect URL to an access denied URL for OIDC. I'm not sure why we do that. Do you remember why @zachmargolis ?
  
modifying the redirect URI  -- that's for cancel links (cancel = access denied)

This is strictly *not* a cancel thing, this is a URL to help people get back to the SPs that directed them to us, so raw `return_to_sp_url` is probably the best bet, [per our documentation](https://developers.login.gov/register/#return-to-sp-urlhttps://developers.login.gov/register/#return-to-sp-url):

> **Return to SP URL**
> The URL of the SP which login.gov provides to users when they wish to go directly to the SP site (or cancel)